logic/src/androidMain/kotlin/com/wire/kalium/logic/feature/call/callback/ParticipantChangedHandlerImpl.kt

```diff
@@ -0,0 +1,55 @@
+package com.wire.kalium.logic.feature.call.callback
+
+import com.sun.jna.Pointer
+import com.wire.kalium.calling.callbacks.ParticipantChangedHandler
+import com.wire.kalium.logic.data.call.AvsClient
+import com.wire.kalium.logic.data.call.AvsClientList
+import com.wire.kalium.logic.data.call.Participant
+import com.wire.kalium.logic.data.id.QualifiedID
+import com.wire.kalium.logic.util.getDomain
+import com.wire.kalium.logic.util.removeDomain
+import org.json.JSONObject
+
+class ParticipantChangedHandlerImpl(
+    private val onParticipantsChanged: (conversationId: String, participants: List<Participant>, clients: AvsClientList) -> Unit
+) : ParticipantChangedHandler {
+
+    override fun onParticipantChanged(conversationId: String, data: String, arg: Pointer?) {
+        val participantsChange = JSONObject(data)
+        val members = participantsChange.getJSONArray("members")
+
+        val participants = mutableListOf<Participant>()
+        val clients = mutableListOf<AvsClient>()
+        for (i in 0 until members.length()) {
+            val member = members.getJSONObject(i)
```

Comment:
You could use `Kotlinx-serialization` here. The rest of the project uses it, and it could be useful for iOS in the future.

Reply:
Indeed we can! I messed up when first working on it and feared `Kotlinx-serialization`